src/geometry/euclidean/circumcenter.lean

```diff
@@ -583,3 +596,75 @@ end
 end simplex
 
 end affine
+
+namespace euclidean_geometry
+
+open affine affine_subspace
+
+variables {V : Type*} {P : Type*} [inner_product_space V] [metric_space P]
+    [normed_add_torsor V P]
+include V
+
+/-- Suppose all distances from `p₁` and `p₂` to the points of a
+simplex are equal, and that `p₁` and `p₂` lie in the affine span of
+`p` with the vertices of that simplex.  Then `p₁` and `p₂` are equal
+or reflections of each other in the affine span of the vertices of the
+simplex. -/
+lemma eq_or_eq_reflection_of_dist_eq {n : ℕ} {s : simplex ℝ P n} {p p₁ p₂ : P} {r : ℝ}
+    (hp₁ : p₁ ∈ affine_span ℝ (insert p (set.range s.points)))
+    (hp₂ : p₂ ∈ affine_span ℝ (insert p (set.range s.points)))
+    (h₁ : ∀ i, dist (s.points i) p₁ = r) (h₂ : ∀ i, dist (s.points i) p₂ = r) :
+  p₁ = p₂ ∨ p₁ = reflection (affine_span ℝ (set.range s.points)) p₂ :=
+begin
+  have h₁' := s.orthogonal_projection_eq_circumcenter_of_dist_eq h₁,
+  have h₂' := s.orthogonal_projection_eq_circumcenter_of_dist_eq h₂,
+  have hn : (affine_span ℝ (set.range s.points) : set P).nonempty :=
+    (affine_span_nonempty ℝ _).2 (set.range_nonempty _),
+  have hc : is_complete ((affine_span ℝ (set.range s.points)).direction : set V) :=
+    submodule.complete_of_finite_dimensional _,
+  rw [←affine_span_insert_affine_span,
+      mem_affine_span_insert_iff (orthogonal_projection_mem hn hc p)] at hp₁ hp₂,
+  rcases hp₁ with ⟨r₁, p₁o, hp₁o, hp₁⟩,
+  rcases hp₂ with ⟨r₂, p₂o, hp₂o, hp₂⟩,
+  obtain rfl : orthogonal_projection (affine_span ℝ (set.range s.points)) p₁ = p₁o,
```

Comment:
Could for consistency us `obtain` for all three lines
```suggestion
  obtain ⟨r₁, p₁o, hp₁o, hp₁⟩ := hp₁,
  obtain ⟨r₂, p₂o, hp₂o, hp₂⟩ := hp₂,
  obtain rfl : orthogonal_projection (affine_span ℝ (set.range s.points)) p₁ = p₁o,
```
but maybe that's not clearer

Reply:
Since `hp₁` and `hp₂` are hypotheses passed to this lemma, and so that would leave them around after deconstructing them, I don't think that's clearer. It might be another matter in the case of a proof that first copied the hypotheses to keep them around after `rcases` (e.g. because the original version was still wanted for passing to other lemmas); `obtain` would then be clearer for the case of deconstructing a hypothesis while keeping the original version around.

Reply:
> and so that would leave them around after deconstructing them

I don't see that behavior when testing locally - in both cases, you're rebinding the name `hp₁` to point to part of the original.

Reply:
In fact it seems `obtain ... := hp₁` deconstructs `hp₁` regardless of whether there is a new hypothesis of that name, like `rcases` but definitely unlike what the documentation of `obtain` says about it being a combination of `have` and `rcases`.